### ISSUE TITLE
Allow features to be removed on build through configuration

### ISSUE THREAD

**Issue by paulboocock** [snowplow/snowplow-javascript-tracker#792]
Once #752 is in place, we can utilise tree shaking to remove dead code. 
If we combine this with tools such as @rollup/plugin-replace then large parts of the code base can be trimmed out when building sp.js for those users who wish to build their own version but with unused features missing.
This was previously suggested in #677, a similar pattern to this should be achievable with rollupjs and rollup plugins.

**Comment by paulboocock** [comment]
This is really a dupe of #450. Closing in favour of the older issue.
